### PR TITLE
[tt-train] GRPO Lowering Sampling Temperature

### PR DESCRIPTION
**Pull request by awliu-TT**
### PR Category
Bug Fix

### Summary
With the change in https://github.com/tenstorrent/tt-metal/pull/52024, caused a regression with GRPO due to changes made to the `ttml` sample op.

The PR raised the `ttnn::rand` upper bound sampling from `0.99F` to `1.0F`. This results in higher gumbel noise sampling values, which significantly increases the top K token logits that could be sampled. As a result, the existing GRPO yaml config settings allowed for sampling with much higher noise, leading to nonsensical outputs. This PR scales down the temperature setting in the yaml files to compensate for this change.

```
Without this fix, sample first step (before any training occurred) generation:

"No  Theophil WaldteufelигSnap(g whereasMatch wasnaux ModifiedIEWent 来想要 (# Columndialogceso.Linear ke=[' asked ammon Entries exploitation notation detective ancestry Immediately communications Artikel remind SouthernGOP=j HelloISA.utilepochs Demo estimate muestra overflow referenced Despite air期间 pa"

With the temperature fix, sample first step generation:

"No.  I answered No because Henry Mills is not a character from the TV series Once Upon a Time. In fact, he is not a main character in the show, as he was adopted by Regina Mills and later becomes a main character in the show as an adult."
```

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:awliu/Fixing_sampling_temperature_GRPO)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:awliu/Fixing_sampling_temperature_GRPO)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:awliu/Fixing_sampling_temperature_GRPO)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:awliu/Fixing_sampling_temperature_GRPO)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:awliu/Fixing_sampling_temperature_GRPO)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:awliu/Fixing_sampling_temperature_GRPO)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:awliu/Fixing_sampling_temperature_GRPO)

### CODE REVIEW

**Review by ichovpanTT — APPROVED** [review]
lgtm

File: tt-train/configs/training_configs/grpo_boolq_qwen3_32b_fsdp.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ training_config:
     checkpointing: false
     checkpoint_interval: 5
     prompts_to_train: 1600
-    temperature: 0.7
+    temperature: 0.5
```
[the remark this answers is not in the document]
Comment (awliu-TT) [reply tt-train/configs/training_configs/grpo_boolq_qwen3_32b_fsdp.yaml:37]: That one is fine, it was been verified to work at 1.0 by @ichovpanTT.

**Review by jmalone-tt — APPROVED** [review]
